libs/community/langchain_community/document_loaders/sharepoint.py

```diff
@@ -37,6 +37,8 @@ class SharePointLoader(O365BaseLoader, BaseLoader):
     """ The ID of the file for which we need auth identities"""
     site_id: Optional[str] = None
     """ The ID of the Sharepoint site of the user where the file is present """
+    load_extended_metadata: Optional[bool] = False
```

Comment:
New to Sharepoint, so might be a basic question: but why would people want this additional metadata?

Reply:
It is an enhancement to the loader. As of today SharepointLoader is not very rich in metadata.
I have added 3 of the fields which makes the Documents metadata ready for consumption for any app.

1. size: size of the source file
2. owner: owner of the source file
3. full_path: pretty human readable path of the source file.

Note: Please refer GoogleDriveLoader with similar signatures: https://github.com/langchain-ai/langchain-google/blob/6495426d7db176fa72803abf9a356d5083bf19c5/libs/community/langchain_google_community/drive.py#L50